### PR TITLE
Add Node.replaceWith capability

### PR DESCRIPTION
**Pull request by kristoferbaxter**
Implements functionality requested in #1023.

### CODE REVIEW

File: src/test/node/replaceWith.test.ts [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  t.context = {
+    document,
+    parent: document.createElement("div"),
+    child: document.createElement("div"),
+    childTwo: document.createElement("div"),
+    childThree: document.createElement("div"),
+  };
+});
+
+test('replacing unattached child results in no changes', (t) => {
```
Comment (samouri) [inline src/test/node/replaceWith.test.ts:42]: One test missing: `replacing child with nothing`. Technically a 0-args call to `replaceWith` should be equivalent to node.remove()

**Review by samouri — CHANGES_REQUESTED** [review]
Great work. One nit about potentially missing edge case

File: src/test/node/replaceWith.test.ts (hunk 1 above)
Comment (kristoferbaxter) [reply src/test/node/replaceWith.test.ts:53]: Added test. Good callout!

**Review by samouri — APPROVED** [review]
